public_html/pipeline.php

```diff
@@ -5,97 +5,78 @@
-    header('Location: /'.$pipeline->name);
-    exit;
+
+# Find release from URL if set
+if(count($path_parts) > 1){
+  foreach($pipeline->releases as $r){
+    if($path_parts[1] == $r->tag_name){
+      $release = $r->tag_name;
+      $release_url = $releases->html_url;
+    }
+  }
+  if($path_parts[1] == 'dev'){
+    $release = 'dev';
+    $release_url = null;
+  }
 }
+
 # Usage docs
-else if($_GET['path'] == $pipeline->name.'/usage'){
-    $pagetab = 'usage';
-    require_once('../includes/pipeline_page/usage.php');
+if(endswith($_GET['path'], '/usage')){
+  $pagetab = 'usage';
+  require_once('../includes/pipeline_page/usage.php');
 }
-# output docs
-else if($_GET['path'] == $pipeline->name.'/output'){
-    $pagetab = 'output';
-    require_once('../includes/pipeline_page/output.php');
+# Output docs
+else if(endswith($_GET['path'], '/output')){
+  $pagetab = 'output';
+  require_once('../includes/pipeline_page/output.php');
 }
 # Stats
-else if($_GET['path'] == $pipeline->name.'/stats'){
-    $pagetab = 'stats';
-    require_once('../includes/pipeline_page/stats.php');
+else if(endswith($_GET['path'], '/stats')){
+  $pagetab = 'stats';
+  require_once('../includes/pipeline_page/stats.php');
 }
 # Releases
-else if($_GET['path'] == $pipeline->name.'/releases'){
-    $pagetab = 'releases';
-    require_once('../includes/pipeline_page/releases.php');
-}
-// handling urls with a version in it
-else if(count($path_parts) >= 2 && preg_match('/^\d+\.\d+\.\d+|^\d+\.\d+$|^dev/',$path_parts[1])){
-  $release = $path_parts[1];
-  foreach($pipeline->releases as $releases)
-  if($releases==$release){
-    $release_url = $releases->html_url;
-  }
-  $release_href = $pipeline->name.'/'.$release;
-  if(count($path_parts) == 2 && preg_match('/^\d+\.\d+\.\d+|^\d+.\d+$|^dev/',$path_parts[1])){
-    $pagetab = 'home';
-    require_once('../includes/pipeline_page/docs.php');
-  }else{
-
-  switch($path_parts[2]){
-    case 'usage':
-      $pagetab = 'usage';
-      require_once('../includes/pipeline_page/usage.php');
-    break;
-    case 'output':
-      $pagetab = 'output';
-      require_once('../includes/pipeline_page/output.php');
-      break;
-    case 'stats':
-      $pagetab = 'stats';
-      require_once('../includes/pipeline_page/stats.php');
-      break;
-    case 'releases':
-      $pagetab = 'releases';
-      require_once('../includes/pipeline_page/releases.php');
-      break;
-    default:
-      header('HTTP/1.1 404 Not Found');
-      include('404.php');
-      die();
-  }
-
-
-  }
-
+else if(endswith($_GET['path'], '/releases')){
+  $pagetab = 'releases';
+  require_once('../includes/pipeline_page/releases.php');
 }
 # Some other URL pattern that we don't recognise - 404
-else {
-    header('HTTP/1.1 404 Not Found');
-    include('404.php');
-    die();
+else if($_GET['path'] != $pipeline->name && $_GET['path'] != $pipeline->name.'/'.$release){
```

Comment:
why do we need the `if`-clause here, wouldn't we catch more cases if it was just `else`?

Reply:
This `if` clause basically catches the homepage, just in reverse. We could make it just an `else`, but then we'd need another similar-looking `if` clause above to allow the homepage.